moved/src/move_execution/canonical.rs

```diff
@@ -153,13 +166,21 @@ pub(super) fn execute_canonical_transaction(
         }
     };
 
+    let gas_used = total_gas_used(&gas_meter, genesis_config);
+    base_token.refund_gas_cost(
+        &sender_move_address,
+        gas_used - l2_cost,
```

Comment:
I think you mentioned this in the meeting, but I wanted to confirm here that you do need to pass along the price because you need to convert the gas amount into a token amount by multiplying by the price. With the price set to zero then there would be no amount charged or refunded so then the existing tests will pass.